vllm/model_executor/models/chatglm.py

```diff
@@ -306,26 +442,67 @@ def __init__(
                                            config.hidden_size,
                                            quant_config=quant_config)
 
+        vision_config_flag = getattr(config, 'vision_config', None)
+        if vision_config_flag is not None:
+            self.vision_config = Namespace(**config.vision_config)
+            self.vision = EVA2CLIPModel(self.config, quant_config)
+        else:
+            self.vision = None
+
     def forward(
         self,
         input_ids: torch.Tensor,
-        position_ids: torch.Tensor,
+        positions: torch.Tensor,
         kv_caches: List[torch.Tensor],
         attn_metadata: AttentionMetadata,
+        intermediate_tensors: Optional[IntermediateTensors] = None,
+        **kwargs: object,
     ) -> torch.Tensor:
+
         inputs_embeds = self.embedding(input_ids)
+        pixel_values = kwargs.pop("image_embeds", None)
+        if pixel_values is not None and self.vision is not None:
+
+            if isinstance(pixel_values, torch.Tensor):
+                if pixel_values.ndim == 2:
+                    pixel_values = pixel_values
+                else:
+                    pixel_values = torch.concat(list(pixel_values))
+            elif isinstance(pixel_values, list):
+                return torch.concat(pixel_values)
+            else:
+                raise TypeError("""pixel_values must be a torch.Tensor 
+                    or a list of torch.Tensor
+                    """)
+
+            pixel_values = pixel_values.to(dtype=inputs_embeds.dtype)
+            image_embeds = self.vision(pixel_values)
+
+            boi_token_id = self.config.boi_token_id
+            eoi_token_id = self.config.eoi_token_id
+
+            inputs_embeds = merge_glm_vision_embeddings(
+                input_ids=input_ids,
+                inputs_embeds=inputs_embeds,
+                vision_embeddings=image_embeds,
+                boi_token_id=boi_token_id,
+                eoi_token_id=eoi_token_id)
 
         # Run encoder.
         hidden_states = self.encoder(
             hidden_states=inputs_embeds,
-            position_ids=position_ids,
+            position_ids=positions,
             kv_caches=kv_caches,
             attn_metadata=attn_metadata,
         )
         return hidden_states
 
 
-class ChatGLMForCausalLM(nn.Module, SupportsLoRA):
+@MULTIMODAL_REGISTRY.register_image_input_mapper()
```

Comment:
VLLM expected user input image data as `PIL.Image` instead of `Tensors` directly. And `PIL.Image` would be processed to `Tensors` in image_input_mapper.

You might need to implement an input mapper for GLM-4v, because it seems that there is no preprocessor implemented in model repo, the default image_input_mapper may not work with `PIL.Image` inputs here.